### PR TITLE
Fix: V1 tests and ERC4626 Implementation

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
Tests needed to be modified to support the V1 vaults. There was also a bug in how the `mint` function was working in the vault, causing it to mint the wrong amount of shares for a user.

# Changes
- references strategies now instead of relying on the vault directly sending funds out
- fixes `mint` function
- creates a new test file for V1 vault tests.

### CODE REVIEW

**Review by 0xKoda — APPROVED** [review]
Looks good! Just needs linting fix

File: contracts/VaultV1.sol [hunk 1]
```diff
@@ -19,26 +21,26 @@ contract VaultV1 is ERC4626, Ownable {
...
     //the strategy address
-    address public STRATEGY;
+    IStrategy public STRATEGY;
     //the buffer amount (indicates to bot the minimun amount of tokens that can be stored in the vault)
     uint256 public BUFFER;
     //top up amount (adjusted on large withdrawals)
     uint256 public TOP_UP;
 
-    constructor(ERC20 underlying) ERC4626(underlying, "EVault", "EVLT") {
+    constructor(ERC20 underlying, address strategy) ERC4626(underlying, "TracerVault", "TVLT") {
```
Comment (0xKoda) [inline contracts/VaultV1.sol:29]: Should the strat fall under the erc4626 constructor?

**Review by 0xKoda — COMMENTED** [review]
Looks good

File: contracts/VaultV1.sol (hunk 1 above)
Comment (raymogg) [reply contracts/VaultV1.sol:29]: Nah I had a chat to Joey the other day - the ERC4626 spec is changing from a yield bearing vault to just a general vault spec. In this vein I don't think strategies are natively included as part of it.

**Pull request merged**